tests/test_subscription.py

```diff
@@ -0,0 +1,129 @@
+
+def test_subscription_class():
+    subscription = generate_subscription_instance()
+    methods = dir(subscription)
+    assert "create" in methods
+    assert is_str(subscription.api_config.server_key)
+    assert is_str(subscription.api_config.client_key)
+
+def test_subscription_create_subscription():
+    subscription = generate_subscription_instance()
+    parameters = generate_param()
+    response = subscription.create(parameters)
+    global SUBSCRIPTION_ID
+    SUBSCRIPTION_ID = response['id']
+    assert isinstance(response, dict)
+    assert 'id' in response.keys()
+    assert response['status'] == 'active'
+
+def test_subscription_fail_empty_param():
+    subscription = generate_subscription_instance()
+    parameters = None
+    err = ''
+    try:
+        response = subscription.create(parameters)
+    except Exception as e:
+        err = e
+    assert 'MidtransAPIError' in err.__class__.__name__
+    assert '400' in err.message
+    assert 'Bad request' in err.message
+
+def test_subscription_fail_zero_amount():
+    subscription = generate_subscription_instance()
+    parameters = generate_param()
+    parameters['amount'] = 0
+    err = ''
+    try:
+        response = subscription.create(parameters)
+    except Exception as e:
+        err = e
+    assert 'MidtransAPIError' in err.__class__.__name__
+    assert '400' in err.message
+    assert 'subscription.amount must be between 0.01 - 99999999999.00' in err.message
+
+def test_subscription_get_subscription():
+    subscription = generate_subscription_instance()
+    parameters = generate_param()
+    response = subscription.get(SUBSCRIPTION_ID)
+    assert isinstance(response, dict)
+    assert response['id'] == SUBSCRIPTION_ID
+    assert response['status'] == 'active'
+
+def test_subscription_get_subscription_not_found():
+    subscription = generate_subscription_instance()
+    err = ''
+    try:
+        response = subscription.get('123')
+    except Exception as e:
+        err = e
+    assert 'MidtransAPIError' in err.__class__.__name__
+    assert '404' in err.message
+    assert 'Subscription doesn\'t exist.' in err.message
+
+def test_subscription_disable_subscription():
+    subscription = generate_subscription_instance()
+    response = subscription.disable(SUBSCRIPTION_ID)
+    assert isinstance(response, dict)
+    assert response['status_message'] == 'Subscription is updated.'
+    get_subscription = subscription.get(SUBSCRIPTION_ID)
+    assert get_subscription['id'] == SUBSCRIPTION_ID
+    assert get_subscription['status'] == 'inactive'
+
+def test_subscription_enable_subscription():
+    subscription = generate_subscription_instance()
+    response = subscription.enable(SUBSCRIPTION_ID)
+    assert isinstance(response, dict)
+    assert response['status_message'] == 'Subscription is updated.'
+    get_subscription = subscription.get(SUBSCRIPTION_ID)
+    assert get_subscription['id'] == SUBSCRIPTION_ID
+    assert get_subscription['status'] == 'active'
```

Comment:
what is the end result of the tested subscription object? is it active? it should be inactive after this test. to prevent Core API continue to execute subscription for each of the created test subscription object on sbox.

Reply:
make sense, will disable after test